src/Plugin/pathauto/AliasType/RdfEntityAliasType.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_entity\Plugin\pathauto\AliasType;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\pathauto\Plugin\pathauto\AliasType\EntityAliasTypeBase;
+use Drupal\rdf_entity\Entity\Query\Sparql\Query;
+use Drupal\rdf_entity\UriEncoder;
+
+/**
+ * A pathauto alias type plugin for RDF entities.
+ *
+ * @AliasType(
+ *   id = "rdf_entity",
+ *   label = @Translation("Rdf entity"),
+ *   types = {"rdf_entity"},
+ *   provider = "rdf_entity",
+ *   context = {
+ *     "rdf_entity" = @ContextDefinition("entity:rdf_entity")
+ *   }
+ * )
+ */
+class RdfEntityAliasType extends EntityAliasTypeBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEntityTypeId() {
+    return 'rdf_entity';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSourcePrefix() {
+    return '/rdf_entity/';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function batchUpdate($action, &$context) {
+    if (!isset($context['sandbox']['count'])) {
+      $context['sandbox']['count'] = 0;
+    }
+
+    $query = $this->getRdfEntityQuery();
+    $query->addTag('rdf_entity_pathauto_bulk_update');
+
+    switch ($action) {
+      case 'create':
+        // Process RDF entities that are not in the list of URL aliases.
+        $aliased_rdf_entity_ids = $this->getAliasedEntityIds($context['sandbox']);
+        if (!empty($aliased_rdf_entity_ids)) {
+          $query->condition('id', $aliased_rdf_entity_ids, 'NOT IN');
```

Comment:
I think this is dangerous when `$aliased_rdf_entity_ids` grows to hundreds of thousands of items. I'm not sure the SPARQL will support so many items in the query. Overall I think we should go with other "bulk update" strategy here.